### PR TITLE
Custom error views

### PR DESCRIPTION
**Pull request by jafffmen**

### CODE REVIEW

File: app/views/layouts/error.haml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+!!! 5
```
Comment (krzysiek1507) [inline app/views/layouts/error.haml]: third layout in the application?

File: app/views/shared/_head.haml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 %meta{content: "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
 %meta{content: "width=device-width, initial-scale=1.0, maximum-scale=1", name: "viewport"}/
 != meta_data_tags
-= canonical_tag(current_store.url)
+%link{ href: current_store.url, rel: 'canonical' }/
```
Comment (krzysiek1507) [inline app/views/shared/_head.haml]: @jafffmen why did you change it?

Comment (jafffmen) [inline app/views/shared/_head.haml]: @krzysiek1507 because it works exactly the same and for an unknown reason it crashes error views

Comment (krzysiek1507) [inline app/views/shared/_head.haml]: @jafffmen did you check this helper (gem `canonical-rails`). It's not the same all the time

Comment (jafffmen) [inline app/views/shared/_head.haml]: @krzysiek1507 yes I checked it and it crashes on `to_sym` method on `request.params['action']` which are missing on error pages

File: config/environments/development.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@
   config.eager_load = false
 
   # Show full error reports and disable caching.
-  config.consider_all_requests_local       = true
+  config.consider_all_requests_local       = false
```
Comment (krzysiek1507) [inline config/environments/development.rb]: @jafffmen we don't want to handle errors on development

File: app/views/shared/_head.haml (hunk 2 above)
Comment (krzysiek1507) [inline app/views/shared/_head.haml]: @jafffmen so we can't use this gem

**Comment by damianlegawiec** [comment]
@jafffmen `Spree::ErrorsController` -> `ErrorsController` it should be used also outside of spree

**Comment by damianlegawiec** [comment]
@jafffmen also please do rebase with master

**Comment by jafffmen** [comment]
@damianlegawiec @krzysiek1507 done :)

**Comment by damianlegawiec** [comment]
@jafffmen please convert haml -> erb ;)
